install.sh

```diff
@@ -1,90 +1,355 @@
-echo "$(tput bold)$(tput setaf 3)To generate the master key, you can use the utility bundled within \n(https://github.com/juspay/hyperswitch-card-vault)$(tput sgr0)"
-echo "$(tput bold)$(tput setaf 3)If you have cargo installed you can run \n(cargo install --git https://github.com/juspay/hyperswitch-card-vault --bin utils --root . && ./bin/utils master-key && rm ./bin/utils && rmdir ./bin)$(tput sgr0)"
-
-echo "Please input the encrypted master key (optional): "
-read -s MASTER_KEY
+
+echo "Dependency installation completed."
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue}    Checking neccessary permissions${reset}"
+echo "${blue}##########################################${reset}\n"
+echo
+
+check_root_user() {
+  AWS_ARN=$(aws sts get-caller-identity --output json | jq -r .Arn )
+  if [[ $AWS_ARN == *":root"* ]]; then
+    echo "ROOT user is not recommended. Please create a new user with AdministratorAccess and use their Access Token."
+    exit 1
+  fi
+}
+
+REQUIRED_POLICIES=("AdministratorAccess") # Add other necessary policies to this array
+# Check if the current user is a root user
+echo "Checking if the current AWS user is root..."
+(check_root_user) & show_loader "Verifying root user status"
+
+check_iam_policies() {
+  USER_POLICIES=$(aws iam list-attached-role-policies --role-name "$AWS_ROLE" --output json | jq -r '.AttachedPolicies[].PolicyName')
+  for policy in "${REQUIRED_POLICIES[@]}"; do
+    if ! echo "$USER_POLICIES" | grep -q "$policy"; then
+      echo "Required policy $policy is not attached to your user. Please attach this policy."
+      exit 1
+    fi
+  done
+  echo "All necessary permissions are in place."
+}
+
+# Check for specific IAM policies
+echo "Checking for necessary IAM policies..."
+(check_iam_policies) & show_loader "Verifying IAM policies"
+
+echo
+echo "${blue}##########################################${reset}"
+echo "${blue} Configure Credentials of the Application ${reset}"
+echo "${blue}##########################################${reset}"
+echo
+
+# Function to validate the password
+validate_password() {
+    local password=$1
+
+    # Check length (at least 8 characters)
+    if [[ ${#password} -lt 8 ]]; then
+        display_error "Error: Password must be at least 8 characters."
+        return 1
+    fi
+
+    # Check if it starts with an alphabet
+    if [[ ! $password =~ ^[A-Za-z] ]]; then
+        display_error "Error: Password must start with a letter."
+        return 1
+    fi
+
+    # Check for at least one uppercase letter, one lowercase letter, one digit, and one special character
+    if [[ ! $password =~ [A-Z] || ! $password =~ [a-z] || ! $password =~ [0-9]  || ! $password =~ [^A-Za-z0-9] ]]; then
+        display_error "Error: Password must include uppercase and lowercase letters, a number, and a special character."
```

Comment:
by default RDS won't allow passwords with special characters. so this check is invalid

Reply:
The password for the database master user can include any printable ASCII character except /, ', ", @, or a space. I will make appropriate changes in this to identify characters such as /, ', ", @ or space.

Please refer this link.
[Quotas and constraints for Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Limits.html)